sdk/appconfiguration/azure-data-appconfiguration/README.md

```diff
@@ -167,68 +167,123 @@ ConfigurationAsyncClient client = new ConfigurationClientBuilder()
+- [Conditional request a configuration setting][sample_conditional_request]
+
+### Create a configuration client
+Create a configuration client by using `ConfigurationClientBuilder` by passing connection string.
 ```Java
 ConfigurationClient client = new ConfigurationClientBuilder()
         .connectionString(connectionString)
         .buildClient();
-ConfigurationSetting setting = client.setSetting("some_key", "some_label", "some_value");
 ```
 
-### Retrieve a Configuration Setting
+### Create a configuration setting
+
+Create a configuration setting to be stored in the configuration store. There are two ways to store a configuration setting:
+- `addConfigurationSetting` creates a setting only if the setting does not already exist in the store.
+- `setConfigurationSetting` creates a setting if it doesn't exist or overrides an existing setting.
 
-Retrieve a previously stored Configuration Setting by calling getSetting.
 ```Java
-ConfigurationClient client = new ConfigurationClientBuilder()
-        .connectionString(connectionString)
-        .buildClient();
-client.setSetting("some_key", "some_label", "some_value");
-ConfigurationSetting setting = client.getSetting("some_key", "some_label");
+ConfigurationSetting setting = client.setConfigurationSetting("some_key", "some_label", "some_value");
 ```
 
-### Update an existing Configuration Setting
+### Retrieve a configuration setting
 
-Update an existing Configuration Setting by calling setSetting.
+Retrieve a previously stored configuration setting by calling `getConfigurationSetting`.
 ```Java
-ConfigurationClient client = new ConfigurationClientBuilder()
-        .connectionString(connectionString)
-        .buildClient();
-client.setSetting("some_key", "some_label", "some_value");
-ConfigurationSetting setting = client.setSetting("some_key", "some_label", "new_value");
+ConfigurationSetting setting = client.setConfigurationSetting("some_key", "some_label", "some_value");
+ConfigurationSetting retrievedSetting = client.getConfigurationSetting("some_key", "some_label");
+```
+For conditional request,
+```Java
+Response<ConfigurationSetting> settingResponse = client.getConfigurationSettingWithResponse(setting, null, true, Context.NONE);
 ```
 
-### Delete a Configuration Setting
+### Update an existing configuration setting
 
-Delete an existing Configuration Setting by calling deleteSetting.
+Update an existing configuration setting by calling `setConfigurationSetting`.
 ```Java
-ConfigurationClient client = new ConfigurationClientBuilder()
-        .connectionString(connectionString)
-        .buildClient();
-client.setSetting("some_key", "some_label", "some_value");
-ConfigurationSetting setting = client.deleteSetting("some_key", "some_label");
+ConfigurationSetting setting = client.setConfigurationSetting("some_key", "some_label", "some_value");
+ConfigurationSetting updatedSetting = client.setConfigurationSetting("some_key", "some_label", "new_value");
+```
+For conditional request,
```

Comment:
What is a conditional request? I think you should add a bit more context here.

Reply:
updated